crates/subspace-farmer/src/plot.rs

```diff
@@ -637,12 +637,12 @@ impl IndexHashToOffsetDB {
         let start = iter
             .key()
             .map(PieceDistance::from_big_endian)
-            .ok_or_else(|| io::Error::other("Failed to fetch piece range"))?;
+            .unwrap_or(PieceDistance::MIDDLE);
```

Comment:
Maybe it should be an optional range if farmer doesn't store anything yet? Why pretending that it is storing `PieceDistance::MIDDLE` if it doesn't?

Reply:
Hmm, isn't for every value `x` , `x..x` is empty range?

Reply:
Hmm, see why it is incorrect now. Don't have strong preference here, but return type here can be either:
- `Range` with bounds `first_piece_hash..last_piece_hash + 1`
- `Option<RangeInclusive>`, so that when we don't have any pieces stored we would return `None`

Reply:
`Option<RangeInclusive>` feels semantically more correct to me